lib/puppet/type/rds_instance.rb

```diff
@@ -0,0 +1,172 @@
+
+Default: ORCL
+
+Constraints:
+
+Cannot be longer than 8 characters
+SQL Server
+
+Not applicable. Must be null.'
+  end
+
+  newproperty(:region) do
+    desc 'the region in which to launch the instance'
+    validate do |value|
+      fail 'region should not contain spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:db_instance_class) do
+    desc 'the instance class to use for the instance eg. db.m3.medium'
+    validate do |value|
+      fail 'db_instance_class should not contain spaces' if value =~ /\s/
+      fail 'db_instance_class should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:availability_zone) do
+    desc 'the availability zone in which to place the instance'
+    validate do |value|
+      fail 'availability_zone should not contain spaces' if value =~ /\s/
+      fail 'availability_zone should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:engine) do
+    desc 'the type of Database for the instance( mysql, postgres, etc...)'
+    validate do |value|
+      fail 'engine type should not contains spaces' if value =~ /\s/
+      fail 'engine should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:engine_version) do
+    desc 'the version of Database for the instance'
+    validate do |value|
+      fail 'engine_version type should not contains spaces' if value =~ /\s/
+      fail 'engine_version should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:allocated_storage) do
+    desc 'The size of the DB'
+    validate do |value|
+      fail 'allocated_storage type should not contains spaces' if value =~ /\s/
+      fail 'allocated_storage should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:license_model) do
+    desc 'the license for the instance (iValid values: license-included | bring-your-own-license | general-public-license)'
+    validate do |value|
+      fail 'license_model type should not contains spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:storage_type) do
+    desc 'The storage stype for the DB (Valid values: gp | io1  *Note: If you specify io1, you must also include a value for the Iops parameter)'
+    validate do |value|
+      fail 'storage_type type should not contains spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:iops) do
+    desc 'The IOPS stype for the DB.'
+    newvalue(/\d+/)
+  end
+
+  newproperty(:master_username) do
+    desc 'The main user for the DB'
+    validate do |value|
+      fail 'master_username type should not contains spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:master_user_password) do
+    desc 'The main user Password'
+    validate do |value|
+      fail 'master_user_password should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:multi_az) do
+    desc 'Define a multi-az'
+    validate do |value|
+      fail 'multi_az should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:db_subnet_group_name) do
```

Comment:
I think this should just be called subnet to match other resources that do the same. Everything is referenced by name so `group_name` is redundant. and `db` is redundant as it's part of the database type.